test/e2e/framework.go

```diff
@@ -643,33 +643,55 @@ func (data *TestData) deleteTestNamespace(timeout time.Duration) error {
 }
 
 // deployAntreaCommon deploys Antrea using kubectl on the control-plane Node.
-func (data *TestData) deployAntreaCommon(yamlFile string, extraOptions string, waitForAgentRollout bool) error {
+func (data *TestData) deployAntreaCommon(yamlFiles []string, extraOptions string, waitForAgentRollout bool) error {
 	// TODO: use the K8s apiserver when server side apply is available?
 	// See https://kubernetes.io/docs/reference/using-api/api-concepts/#server-side-apply
-	rc, _, _, err := provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl apply %s -f %s", extraOptions, yamlFile))
-	if err != nil || rc != 0 {
-		return fmt.Errorf("error when deploying Antrea; is %s available on the control-plane Node?", yamlFile)
+	deployYML := func(yml string) error {
+		rc, _, _, err := provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl apply %s -f %s", extraOptions, yml))
+		if err != nil || rc != 0 {
+			return fmt.Errorf("error when deploying Antrea; is %s available on the control-plane Node?", yml)
+		}
+		return nil
 	}
-	rc, stdout, stderr, err := provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl -n %s rollout status deploy/%s --timeout=%v", antreaNamespace, antreaDeployment, defaultTimeout))
-	if err != nil || rc != 0 {
-		return fmt.Errorf("error when waiting for antrea-controller rollout to complete - rc: %v - stdout: %v - stderr: %v - err: %v", rc, stdout, stderr, err)
+	rolloutAgent := func(yml string) error {
+		ds := antreaDaemonSet
+		if yml == antreaWindowsYML {
+			ds = antreaWindowsDaemonSet
+		}
+		rc, stdout, stderr, err := provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl -n %s rollout status ds/%s --timeout=%v", antreaNamespace, ds, defaultTimeout))
+		if err != nil || rc != 0 {
+			return fmt.Errorf("error when waiting for antrea-agent (%s) rollout to complete - rc: %v - stdout: %v - stderr: %v - err: %v", yml, rc, stdout, stderr, err)
+		}
+		return nil
+	}
+
+	for _, yamlFile := range yamlFiles {
+		if err := deployYML(yamlFile); err != nil {
+			return err
+		}
 	}
 	if waitForAgentRollout {
-		rc, _, _, err = provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl -n %s rollout status ds/%s --timeout=%v", antreaNamespace, antreaDaemonSet, defaultTimeout))
-		if err != nil || rc != 0 {
-			return fmt.Errorf("error when waiting for antrea-agent rollout to complete - rc: %v - stdout: %v - stderr: %v - err: %v", rc, stdout, stderr, err)
+		for _, yamlFile := range yamlFiles {
+			if err := rolloutAgent(yamlFile); err != nil {
```

Comment:
this is weird to me because it implies that all the YAML files define an antrea-agent DaemonSet. This limits how you can use `deployAntreaOption`.

Reply:
Make sense. I have refactored these code.